scripts/vendor-sqlite3.swift

```diff
@@ -0,0 +1,372 @@
+        defer { group.leave() }
+        if let error = error {
+            getError = error
+            return
+        }
+        guard let data = data else {
+            getError = "No data returned from \(url)"
+            return
+        }
+        getData = data
+    }.resume()
+    group.wait()
+    if let getError = getError {
+        throw getError
+    }
+    return getData!
+}
+
+func downloadAndUnzipSQLite(from url: URL, to filename: String, expectedSize: Int?, expectedSha3Sum: String?) throws {
+    let productData = try httpGet(from: url)
+    if let expectedSize = expectedSize {
+        guard productData.count == expectedSize else {
+            throw "Downloaded SQLite archive size \(productData.count) does not match expected \(expectedSize)"
+        }
+    }
+    let file = URL(fileURLWithPath: FileManager.default.currentDirectoryPath)
+        .appendingPathComponent(filename)
+    let directory = file.deletingPathExtension()
+    try productData.write(to: file)
+    if let expectedSha3Sum = expectedSha3Sum {
+        let sha3Output = try subprocess(sha3sum, ["-z", "-a", "256", file.path], captureStdout: true)
+        guard let sha3AndFile = sha3Output?.split(separator: " "),
+              sha3AndFile.count == 2 else {
+            throw "Unknown sha3 sum: \(String(describing: sha3Output))"
+        }
+        guard expectedSha3Sum == sha3AndFile[0] else {
+            throw "Unexpected sha3: \(expectedSha3Sum) != \(sha3AndFile[0])"
+        }
+    }
+    defer {
+        try? FileManager.default.removeItem(at: file)
+        try? FileManager.default.removeItem(at: directory)
+    }
+
+    try subprocess(unzip, [file.path], captureStdout: false)
+
+    _ = try FileManager.default.replaceItemAt(sqlite3Source, withItemAt: directory.appendingPathComponent("sqlite3.c"))
+    _ = try FileManager.default.replaceItemAt(sqlite3Header, withItemAt: directory.appendingPathComponent("sqlite3.h"))
+}
+
+@discardableResult
+func subprocess(_ executable: URL, _ arguments: [String], captureStdout: Bool) throws -> String? {
+    guard FileManager.default.fileExists(atPath: executable.path) else {
+        throw "This script requires \(executable.path), please install it and try again"
+    }
+
+    let stdout = Pipe()
+    let process = Process()
+    process.executableURL = executable
+    process.arguments = arguments
+    if captureStdout {
+        process.standardOutput = stdout
+    }
+    try process.run()
+    process.waitUntilExit()
+    guard process.terminationStatus == 0 else {
+        throw "\(executable.path) failed: \(process.terminationStatus)"
+    }
+    if captureStdout {
+        return String(
+            data: stdout.fileHandleForReading.readDataToEndOfFile(),
+            encoding: .utf8
+        )?.trimmingCharacters(in: .whitespacesAndNewlines)
+    } else {
+        return nil
+    }
+}
+
+func ensureExecutable(_ executable: String) throws -> URL {
+    let bash = URL(fileURLWithPath: "/bin/bash")
```

Comment:
This is a bad assumption - users with path configuration in startup scripts will have problems if they use zsh, for example. You want to use the `PATH` that's in use in the script's environment already - and the easy way is to just execute `/usr/bin/which` directly instead of indirecting through a shell; a standalone version exists for just this reason 🙂.

Reply:
Come to think of it, you could get _really_ fancy and do `/usr/bin/env which ...` so that alternate versions of `which` can also take effect 🤣

Reply:
Totally fair, I've switched to using `/usr/bin/env`